### DISCUSSION TITLE
Issue with next/jest

### DISCUSSION THREAD

**Discussion by ryanhaticus** [vercel/next.js#38353] [Help] [answered]
For some reason, my environment variables aren't being "loaded" into process.env by `next/jest`.
Here's my configuration and an example test.

Configuration:
```ts
import NextJest from 'next/jest';

const createJestConfig = NextJest({
  dir: './',
});

const customJestConfig = {
  moduleDirectories: ['node_modules', '<rootDir>/'],
  testEnvironment: 'jest-environment-jsdom',
};

module.exports = createJestConfig(customJestConfig);
```

Test:
```ts
describe('HTML document root', () => {
  it('Should have a language environment variable available', () => {
    expect(process.env.NEXT_PUBLIC_DOCUMENT_LANG).toBeDefined();
  });
});

export {};
```

I have a TypeScript basePath of `.` (root).

**Comment by Yofou** [comment, marked as answer]
This could be a just a hunch but maybe you need to use `.env.test.local` for it to be loaded properly

**Reply by Yofou** [reply]
I would look at this, https://stackoverflow.com/a/67997819

**Reply by ryanhaticus** [reply]
Hey there! So the correct solution came from the StackOverflow post. I'm going to repost my implementation here for redundancy.
**Please read the included comment.**

```ts
import { loadEnvConfig } from '@next/env';
import fs from 'fs';

// I copy .env.development to .env.test as loadEnvConfig() will only load the environment variables for the testing environment.
// This is intended behavior thought it may not be ideal for all developers. In my case, I just add `.env*.test` to my `.gitignore` and all is well.

const envFile = fs.readFileSync('.env.development', 'utf8');
fs.writeFileSync('.env.test', envFile);

const projectDir = process.cwd();
loadEnvConfig(projectDir);
```

I put this at the top of my `jest.config.ts` file (`jest.config.js` for my JavaScript folks). I may open a PR to add a hyperlink to this documentation. I have seen similar issues and discussion posts regarding my issue.

**Reply by ryanhaticus** [reply]
Please refer to pull request #38359
